nntrainer/graph/network_graph.cpp

```diff
@@ -85,14 +85,26 @@ int NetworkGraph::compile(const std::string &loss_type) {
 
 void NetworkGraph::setExecutionOrder() {
   auto max_count = graph.size() * 3;
-  /** @todo: remove backwarding count for non-trainble layers */
+  size_t not_trainables = 0;
+
+  for (auto iter = cbegin(); iter != cend(); iter++) {
+    if (!(*iter)->getTrainable())
+      not_trainables++;
+  }
+
   for (auto iter = cbegin(); iter != cend(); iter++) {
```

Comment:
How about we iterate from crbegin to crend? IMO iterate from end to begin will make logic more simple.
Please check https://gist.github.com/lhs8928/5610124f0de7787d1eb1b7248159f864